File: lib/spiral/crypto_contract.ts
import { Record } from "../../deps/Fable/src/fable-library-ts/Types.js";
import { int32 } from "../../deps/Fable/src/fable-library-ts/Int32.js";
import { IComparable, IEquatable } from "../../deps/Fable/src/fable-library-ts/Util.js";
import { record_type, int32_type, TypeInfo } from "../../deps/Fable/src/fable-library-ts/Reflection.js";
import { value, Option } from "../../deps/Fable/src/fable-library-ts/Option.js";
import * as crypto from "crypto";

export interface ICryptoCreateHash {
    createHash(x: string): any
}

export class Mut0 extends Record implements IEquatable<Mut0>, IComparable<Mut0> {
    l0: int32;
    constructor(l0: int32) {
        super();
        this.l0 = (l0 | 0);
    }
}

export function Mut0_$reflection(): TypeInfo {
    return record_type("Crypto.Mut0", [], Mut0, () => [["l0", int32_type]]);
}

export function method0(v0_1: int32, v1: Mut0): boolean {
    return v1.l0 < v0_1;
}

export function method1(): string {
    return "";
}

export function closure0(unitVar: void, v0_1: string): string {
    let _v1: Option<string> = void 0;
    const v29: ICryptoCreateHash = crypto;
    const v32: any = v29.createHash("sha256");
    const v34: any = v32.update(v0_1, 'utf8');
    const x: string = v34.digest("hex");
    _v1 = x;
    return value(_v1);
}

export const v0 = (v: string): string => closure0(void 0, v);

export function hash_text(x: string): string {
    return v0(x);
}

File: lib/spiral/crypto_contract.py
from __future__ import annotations
from abc import abstractmethod
from collections.abc import Callable
from dataclasses import dataclass
from typing import (Any, Protocol)
from fable_modules.fable_library.option import value as value_1
from fable_modules.fable_library.reflection import (TypeInfo, int32_type, record_type)
from fable_modules.fable_library.types import Record

class ICryptoCreateHash(Protocol):
    @abstractmethod
    def create_hash(self, x: str) -> Any:
        ...


def _expr0() -> TypeInfo:
    return record_type("Crypto.Mut0", [], Mut0, lambda: [("l0", int32_type)])


@dataclass(eq = False, repr = False, slots = True)
class Mut0(Record):
    l0: int

Mut0_reflection = _expr0

def method0(v0_1: int, v1: Mut0) -> bool:
    return v1.l0 < v0_1


def method1(__unit: None=None) -> str:
    return ""


def closure0(unit_var: None, v0_1: str) -> str:
    _v1: str | None = None
    _v1 = None
    return value_1(_v1)


def _arrow11(v: str) -> str:
    return closure0(None, v)


v0: Callable[[str], str] = _arrow11

def hash_text(x: str) -> str:
    return v0(x)


